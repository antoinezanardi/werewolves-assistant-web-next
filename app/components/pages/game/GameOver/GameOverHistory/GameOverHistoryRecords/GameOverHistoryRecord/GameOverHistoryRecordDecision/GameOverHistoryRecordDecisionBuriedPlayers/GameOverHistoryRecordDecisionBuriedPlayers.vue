<template>
  <div
    id="game-over-history-record-decision-buried-players"
    class="flex flex-col gap-4 items-center justify-center"
  >
    <h4
      v-if="buriedPlayerRoleNameLabel"
      id="game-over-history-record-decision-buried-player-role-name"
      class="text-center"
    >
      {{ buriedPlayerRoleNameLabel }}
    </h4>

    <ul class="mx-auto overflow-x-auto scrollbar-thin text-center w-full whitespace-nowrap">
      <GameOverHistoryRecordDecisionBuriedPlayer
        v-for="buriedPlayer in buriedPlayers"
        :key="buriedPlayer._id"
        :buried-player="buriedPlayer"
        class="game-over-history-record-buried-player inline-block"
        :game-history-record="props.gameHistoryRecord"
        is-disabled
      />
    </ul>
  </div>
</template>

<script setup lang="ts">
import type { GameOverHistoryRecordDecisionBuriedPlayersProps } from "~/components/pages/game/GameOver/GameOverHistory/GameOverHistoryRecords/GameOverHistoryRecord/GameOverHistoryRecordDecision/GameOverHistoryRecordDecisionBuriedPlayers/game-over-history-record-decision-buried-players.types";
import GameOverHistoryRecordDecisionBuriedPlayer from "~/components/pages/game/GameOver/GameOverHistory/GameOverHistoryRecords/GameOverHistoryRecord/GameOverHistoryRecordDecision/GameOverHistoryRecordDecisionBuriedPlayers/GameOverHistoryRecordDecisionBuriedPlayer/GameOverHistoryRecordDecisionBuriedPlayer.vue";
import type { Player } from "~/composables/api/game/types/players/player.class";
import { useRoleName } from "~/composables/api/role/useRoleName";

const props = defineProps<GameOverHistoryRecordDecisionBuriedPlayersProps>();

const { getDefiniteRoleNameLabel } = useRoleName();

const buriedPlayers = computed<Player[]>(() => {
  const { interactions } = props.gameHistoryRecord.play.source;
  const buryDeadBodiesInteraction = interactions?.find(({ type }) => type === "bury");

  return buryDeadBodiesInteraction?.eligibleTargets ?? [];
});

const isBuriedPlayerAlone = computed<boolean>(() => buriedPlayers.value.length === 1);

const buriedPlayerRoleNameLabel = computed<string | undefined>(() => {
  if (!isBuriedPlayerAlone.value) {
    return "";
  }
  const buriedPlayer = buriedPlayers.value[0];

  return getDefiniteRoleNameLabel(buriedPlayer.role.current, 1);
});
</script>