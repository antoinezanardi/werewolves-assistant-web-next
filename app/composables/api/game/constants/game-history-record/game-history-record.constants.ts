const GAME_HISTORY_RECORD_PLAY_VOTING_RESULTS = [
  "sheriff-election",
  "tie",
  "death",
  "inconsequential",
  "skipped",
] as const;

const GAME_HISTORY_RECORD_PLAYER_ATTRIBUTE_ALTERATION_STATUSES = [
  "attached",
  "detached",
  "activated",
] as const;

export {
  GAME_HISTORY_RECORD_PLAY_VOTING_RESULTS,
  GAME_HISTORY_RECORD_PLAYER_ATTRIBUTE_ALTERATION_STATUSES,
};