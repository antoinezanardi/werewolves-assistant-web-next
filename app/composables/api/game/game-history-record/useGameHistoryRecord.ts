import type { GameHistoryRecord } from "~/composables/api/game/types/game-history-record/game-history-record.class";

type UseGameHistoryRecord = {
  didSourceSkipped: (gameHistoryRecord: GameHistoryRecord) => boolean;
};

function useGameHistoryRecord(): UseGameHistoryRecord {
  function didSourceSkipped(gameHistoryRecord: GameHistoryRecord): boolean {
    const {
      type,
      votes,
      targets,
      didJudgeRequestAnotherVote,
      chosenCard,
    } = gameHistoryRecord.play;

    return type === "vote" && !votes ||
      type === "target" && !targets ||
      type === "request-another-vote" && didJudgeRequestAnotherVote === undefined ||
      type === "choose-card" && !chosenCard;
  }
  return { didSourceSkipped };
}

export { useGameHistoryRecord };