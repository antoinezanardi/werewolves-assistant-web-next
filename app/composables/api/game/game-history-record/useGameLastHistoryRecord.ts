import type { MaybeRef, Ref } from "vue";
import type { GameHistoryRecordPlayerAttributeAlterationStatus } from "~/composables/api/game/types/game-history-record/game-history-record-player-attribute-alteration/game-history-record-player-attribute-alteration.types";
import type { GameHistoryRecord } from "~/composables/api/game/types/game-history-record/game-history-record.class";
import type { Game } from "~/composables/api/game/types/game.class";
import type { GameSource } from "~/composables/api/game/types/game.types";
import type { PlayerAttributeName } from "~/composables/api/game/types/players/player-attribute/player-attribute.types";
import type { Player } from "~/composables/api/game/types/players/player.class";

type UseGameLastHistoryRecord = {
  lastTargetedPlayers: Ref<Player[]>;
  doesHavePlayerAttributeAlteration: (attributeName: PlayerAttributeName, source: GameSource, status: GameHistoryRecordPlayerAttributeAlterationStatus) => boolean;
};

function useGameLastHistoryRecord(game: MaybeRef<Game>): UseGameLastHistoryRecord {
  const lastGameHistoryRecord = computed<GameHistoryRecord | null>(() => (isRef(game) ? game.value.lastGameHistoryRecord : game.lastGameHistoryRecord));

  const lastTargetedPlayers = computed<Player[]>(() => {
    if (!lastGameHistoryRecord.value?.play.targets) {
      return [];
    }
    return lastGameHistoryRecord.value.play.targets.map(target => target.player);
  });

  function doesHavePlayerAttributeAlteration(attributeName: PlayerAttributeName, source: GameSource, status: GameHistoryRecordPlayerAttributeAlterationStatus): boolean {
    if (!lastGameHistoryRecord.value?.playerAttributeAlterations) {
      return false;
    }
    return lastGameHistoryRecord.value.playerAttributeAlterations.some(playerAttributeAlteration => playerAttributeAlteration.name === attributeName &&
      playerAttributeAlteration.source === source &&
      playerAttributeAlteration.status === status);
  }
  return {
    lastTargetedPlayers,
    doesHavePlayerAttributeAlteration,
  };
}

export { useGameLastHistoryRecord };