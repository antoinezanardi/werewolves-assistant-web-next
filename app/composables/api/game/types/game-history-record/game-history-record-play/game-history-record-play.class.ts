import { Expose, plainToInstance, Type } from "class-transformer";

import { GameAdditionalCard } from "~/composables/api/game/types/game-additional-card/game-additional-card.class";
import { GameHistoryRecordPlaySource } from "~/composables/api/game/types/game-history-record/game-history-record-play/game-history-record-play-source/game-history-record-play-source.class";
import { GameHistoryRecordPlayTarget } from "~/composables/api/game/types/game-history-record/game-history-record-play/game-history-record-play-target/game-history-record-play-target.class";
import { GameHistoryRecordPlayVote } from "~/composables/api/game/types/game-history-record/game-history-record-play/game-history-record-play-vote/game-history-record-play-vote.class";
import { GameHistoryRecordPlayVoting } from "~/composables/api/game/types/game-history-record/game-history-record-play/game-history-record-play-voting/game-history-record-play-voting.class";
import type { GamePlayAction, GamePlayCause, GamePlayType } from "~/composables/api/game/types/game-play/game-play.types";
import type { RoleSide } from "~/composables/api/role/types/role.types";
import { DEFAULT_PLAIN_TO_INSTANCE_OPTIONS } from "~/utils/constants/class-transformer.constants";

class GameHistoryRecordPlay {
  @Expose()
  public type: GamePlayType;

  @Expose()
  public action: GamePlayAction;

  @Expose()
  @Type(() => GameHistoryRecordPlaySource)
  public source: GameHistoryRecordPlaySource;

  @Expose()
  public cause?: GamePlayCause;

  @Expose()
  @Type(() => GameHistoryRecordPlayTarget)
  public targets?: GameHistoryRecordPlayTarget[];

  @Expose()
  @Type(() => GameHistoryRecordPlayVote)
  public votes?: GameHistoryRecordPlayVote[];

  @Expose()
  @Type(() => GameHistoryRecordPlayVoting)
  public voting?: GameHistoryRecordPlayVoting;

  @Expose()
  public didJudgeRequestAnotherVote?: boolean;

  @Expose()
  @Type(() => GameAdditionalCard)
  public chosenCard?: GameAdditionalCard;

  @Expose()
  public chosenSide?: RoleSide;

  public static create(gameHistoryRecordPlay: GameHistoryRecordPlay): GameHistoryRecordPlay {
    return plainToInstance(GameHistoryRecordPlay, gameHistoryRecordPlay, DEFAULT_PLAIN_TO_INSTANCE_OPTIONS);
  }
}

export { GameHistoryRecordPlay };