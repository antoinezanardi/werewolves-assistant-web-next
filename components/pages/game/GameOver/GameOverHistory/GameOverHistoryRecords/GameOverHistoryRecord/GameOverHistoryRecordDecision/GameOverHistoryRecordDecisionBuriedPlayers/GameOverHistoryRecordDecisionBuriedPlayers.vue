<template>
  <div
    id="game-over-history-record-decision-buried-players"
    class="flex gap-4 items-center justify-center"
  >
    <GameOverHistoryRecordDecisionBuriedPlayer
      v-for="buriedPlayer in truncatedBuriedPlayers"
      :key="buriedPlayer._id"
      :buried-player="buriedPlayer"
      class="game-over-history-record-buried-player w-full"
      :game-history-record="props.gameHistoryRecord"
    />

    <OverflowTag
      id="buried-players-overflow-tag"
      :entities-count="buriedPlayers.length"
      :maximum-entities-displayed="maximumBuriedPlayersDisplayed"
    />
  </div>
</template>

<script setup lang="ts">
import type { GameOverHistoryRecordDecisionBuriedPlayersProps } from "~/components/pages/game/GameOver/GameOverHistory/GameOverHistoryRecords/GameOverHistoryRecord/GameOverHistoryRecordDecision/GameOverHistoryRecordDecisionBuriedPlayers/game-over-history-record-decision-buried-players.types";
import GameOverHistoryRecordDecisionBuriedPlayer from "~/components/pages/game/GameOver/GameOverHistory/GameOverHistoryRecords/GameOverHistoryRecord/GameOverHistoryRecordDecision/GameOverHistoryRecordDecisionBuriedPlayers/GameOverHistoryRecordDecisionBuriedPlayer/GameOverHistoryRecordDecisionBuriedPlayer.vue";
import OverflowTag from "~/components/shared/misc/OverflowTag/OverflowTag.vue";
import type { Player } from "~/composables/api/game/types/players/player.class";

const props = defineProps<GameOverHistoryRecordDecisionBuriedPlayersProps>();

const maximumBuriedPlayersDisplayed = 3;

const buriedPlayers = computed<Player[]>(() => {
  const { interactions } = props.gameHistoryRecord.play.source;
  const buryDeadBodiesInteraction = interactions?.find(({ type }) => type === "bury");

  return buryDeadBodiesInteraction?.eligibleTargets ?? [];
});

const truncatedBuriedPlayers = computed<Player[]>(() => buriedPlayers.value.slice(0, maximumBuriedPlayersDisplayed));
</script>