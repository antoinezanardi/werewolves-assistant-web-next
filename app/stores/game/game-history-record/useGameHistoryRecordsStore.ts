import type { AsyncDataRequestStatus } from "nuxt/app";
import { defineStore } from "pinia";

import { useFetchGameHistoryRecords } from "~/composables/api/game/game-history-record/useFetchGameHistoryRecords";
import type { GameHistoryRecord } from "~/composables/api/game/types/game-history-record/game-history-record.class";
import { StoreIds } from "~/stores/enums/store.enum";

const useGameHistoryRecordsStore = defineStore(StoreIds.GAME_HISTORY_RECORDS, () => {
  const { getGameHistoryRecords } = useFetchGameHistoryRecords();

  const gameHistoryRecords = ref<GameHistoryRecord[]>([]);
  const fetchingGameHistoryRecordsStatus = ref<AsyncDataRequestStatus>("idle");

  async function fetchAndSetGameHistoryRecords(gameId: string): Promise<void> {
    fetchingGameHistoryRecordsStatus.value = "pending";
    const fetchedGameHistoryRecords = await getGameHistoryRecords(gameId);
    if (!fetchedGameHistoryRecords) {
      fetchingGameHistoryRecordsStatus.value = "error";

      return;
    }
    gameHistoryRecords.value = fetchedGameHistoryRecords;
    fetchingGameHistoryRecordsStatus.value = "success";
  }
  return {
    gameHistoryRecords,
    fetchingGameHistoryRecordsStatus,
    fetchAndSetGameHistoryRecords,
  };
});

export { useGameHistoryRecordsStore };