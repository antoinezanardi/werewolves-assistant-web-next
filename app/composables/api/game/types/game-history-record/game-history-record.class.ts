import { Expose, plainToInstance, Type } from "class-transformer";

import { GameHistoryRecordPlay } from "~/composables/api/game/types/game-history-record/game-history-record-play/game-history-record-play.class";
import { GameHistoryRecordPlayerAttributeAlteration } from "~/composables/api/game/types/game-history-record/game-history-record-player-attribute-alteration/game-history-record-player-attribute-alteration.class";
import { GamePhase } from "~/composables/api/game/types/game-phase/game-phase.class";
import { Player } from "~/composables/api/game/types/players/player.class";
import { DEFAULT_PLAIN_TO_INSTANCE_OPTIONS } from "~/utils/constants/class-transformer.constants";

class GameHistoryRecord {
  @Expose()
  public _id: string;

  @Expose()
  public gameId: string;

  @Expose()
  public turn: number;

  @Type(() => GamePhase)
  @Expose()
  public phase: GamePhase;

  @Expose()
  public tick: number;

  @Expose()
  @Type(() => GameHistoryRecordPlay)
  public play: GameHistoryRecordPlay;

  @Expose()
  @Type(() => Player)
  public revealedPlayers?: Player[];

  @Expose()
  @Type(() => Player)
  public deadPlayers?: Player[];

  @Expose()
  @Type(() => GameHistoryRecordPlayerAttributeAlteration)
  public playerAttributeAlterations?: GameHistoryRecordPlayerAttributeAlteration[];

  @Expose()
  public createdAt: Date;

  public static create(gameHistoryRecord: GameHistoryRecord): GameHistoryRecord {
    return plainToInstance(GameHistoryRecord, gameHistoryRecord, DEFAULT_PLAIN_TO_INSTANCE_OPTIONS);
  }
}

export { GameHistoryRecord };