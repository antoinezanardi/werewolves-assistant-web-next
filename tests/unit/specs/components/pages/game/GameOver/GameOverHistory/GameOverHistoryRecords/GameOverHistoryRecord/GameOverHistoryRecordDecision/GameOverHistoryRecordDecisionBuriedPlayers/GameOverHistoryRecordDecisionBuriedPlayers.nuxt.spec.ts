import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import type { GameOverHistoryRecordDecisionBuriedPlayersProps } from "~/components/pages/game/GameOver/GameOverHistory/GameOverHistoryRecords/GameOverHistoryRecord/GameOverHistoryRecordDecision/GameOverHistoryRecordDecisionBuriedPlayers/game-over-history-record-decision-buried-players.types";
import type GameOverHistoryRecordDecisionBuriedPlayer from "~/components/pages/game/GameOver/GameOverHistory/GameOverHistoryRecords/GameOverHistoryRecord/GameOverHistoryRecordDecision/GameOverHistoryRecordDecisionBuriedPlayers/GameOverHistoryRecordDecisionBuriedPlayer/GameOverHistoryRecordDecisionBuriedPlayer.vue";
import GameOverHistoryRecordDecisionBuriedPlayers from "~/components/pages/game/GameOver/GameOverHistory/GameOverHistoryRecords/GameOverHistoryRecord/GameOverHistoryRecordDecision/GameOverHistoryRecordDecisionBuriedPlayers/GameOverHistoryRecordDecisionBuriedPlayers.vue";
import type OverflowTag from "~/components/shared/misc/OverflowTag/OverflowTag.vue";
import type { Player } from "~/composables/api/game/types/players/player.class";
import { createFakeGameHistoryRecordPlaySource } from "~/tests/unit/utils/factories/composables/api/game/game-history-record/game-history-record-play/game-history-record-play-source/game-history-record-play-source.factory";
import { createFakeGameHistoryRecordPlay } from "~/tests/unit/utils/factories/composables/api/game/game-history-record/game-history-record-play/game-history-record-play.factory";
import { createFakeGameHistoryRecord } from "~/tests/unit/utils/factories/composables/api/game/game-history-record/game-history-record.factory";
import { createFakeGamePlaySourceInteraction } from "~/tests/unit/utils/factories/composables/api/game/game-play/game-play-source/game-play-source-interaction/game-play-source-interaction.factory";
import { createFakeActorAlivePlayer } from "~/tests/unit/utils/factories/composables/api/game/player/player-with-role.factory";

import { mountSuspendedComponent } from "~/tests/unit/utils/helpers/mount.helpers";

describe("Game Over History Record Decision Buried Players Component", () => {
  let wrapper: ReturnType<typeof mount<typeof GameOverHistoryRecordDecisionBuriedPlayers>>;
  const defaultBuriedPlayers = [
    createFakeActorAlivePlayer({ name: "Antoine" }),
    createFakeActorAlivePlayer({ name: "Vanessa" }),
    createFakeActorAlivePlayer({ name: "Thomas" }),
    createFakeActorAlivePlayer({ name: "Doudou" }),
  ];
  const defaultGameHistoryRecord = createFakeGameHistoryRecord({
    play: createFakeGameHistoryRecordPlay({
      type: "bury-dead-bodies",
      action: "bury-dead-bodies",
      source: createFakeGameHistoryRecordPlaySource({
        interactions: [
          createFakeGamePlaySourceInteraction({
            type: "bury",
            eligibleTargets: defaultBuriedPlayers,
          }),
          createFakeGamePlaySourceInteraction({
            type: "steal-role",
            eligibleTargets: [
              createFakeActorAlivePlayer({ name: "Antoine" }),
              createFakeActorAlivePlayer({ name: "Vanessa" }),
            ],
          }),
        ],
      }),
    }),
  });
  const defaultProps: GameOverHistoryRecordDecisionBuriedPlayersProps = { gameHistoryRecord: defaultGameHistoryRecord };

  async function mountGameOverHistoryRecordDecisionBuriedPlayersComponent(options: ComponentMountingOptions<typeof GameOverHistoryRecordDecisionBuriedPlayers> = {}):
  Promise<ReturnType<typeof mount<typeof GameOverHistoryRecordDecisionBuriedPlayers>>> {
    return mountSuspendedComponent(GameOverHistoryRecordDecisionBuriedPlayers, {
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountGameOverHistoryRecordDecisionBuriedPlayersComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Buried Players", () => {
    it("should display the truncated list of buried players when rendered.", () => {
      const buriedPlayers = wrapper.findAllComponents<typeof GameOverHistoryRecordDecisionBuriedPlayer>(".game-over-history-record-buried-player");

      expect(buriedPlayers).toHaveLength(3);
      expect(buriedPlayers[0].props("buriedPlayer")).toStrictEqual<Player>(defaultBuriedPlayers[0]);
      expect(buriedPlayers[1].props("buriedPlayer")).toStrictEqual<Player>(defaultBuriedPlayers[1]);
      expect(buriedPlayers[2].props("buriedPlayer")).toStrictEqual<Player>(defaultBuriedPlayers[2]);
    });

    it("should display all buried players when there are less than 4 buried players.", async() => {
      wrapper = await mountGameOverHistoryRecordDecisionBuriedPlayersComponent({
        props: {
          gameHistoryRecord: createFakeGameHistoryRecord({
            play: createFakeGameHistoryRecordPlay({
              type: "bury-dead-bodies",
              action: "bury-dead-bodies",
              source: createFakeGameHistoryRecordPlaySource({
                interactions: [
                  createFakeGamePlaySourceInteraction({
                    type: "bury",
                    eligibleTargets: defaultBuriedPlayers.slice(0, 2),
                  }),
                  createFakeGamePlaySourceInteraction({
                    type: "steal-role",
                    eligibleTargets: [
                      createFakeActorAlivePlayer({ name: "Antoine" }),
                      createFakeActorAlivePlayer({ name: "Vanessa" }),
                    ],
                  }),
                ],
              }),
            }),
          }),
        },
      });
      const buriedPlayers = wrapper.findAllComponents<typeof GameOverHistoryRecordDecisionBuriedPlayer>(".game-over-history-record-buried-player");

      expect(buriedPlayers).toHaveLength(2);
      expect(buriedPlayers[0].props("buriedPlayer")).toStrictEqual<Player>(defaultBuriedPlayers[0]);
      expect(buriedPlayers[1].props("buriedPlayer")).toStrictEqual<Player>(defaultBuriedPlayers[1]);
    });

    it("should display no buried players when there are no interactions.", async() => {
      wrapper = await mountGameOverHistoryRecordDecisionBuriedPlayersComponent({
        props: {
          gameHistoryRecord: createFakeGameHistoryRecord({
            play: createFakeGameHistoryRecordPlay({
              type: "bury-dead-bodies",
              action: "bury-dead-bodies",
              source: createFakeGameHistoryRecordPlaySource(),
            }),
          }),
        },
      });
      const buriedPlayers = wrapper.findAllComponents<typeof GameOverHistoryRecordDecisionBuriedPlayer>(".game-over-history-record-buried-player");

      expect(buriedPlayers).toHaveLength(0);
    });

    it("should display no buried players when there are no eligible targets.", async() => {
      wrapper = await mountGameOverHistoryRecordDecisionBuriedPlayersComponent({
        props: {
          gameHistoryRecord: createFakeGameHistoryRecord({
            play: createFakeGameHistoryRecordPlay({
              type: "bury-dead-bodies",
              action: "bury-dead-bodies",
              source: createFakeGameHistoryRecordPlaySource({ interactions: [createFakeGamePlaySourceInteraction({ type: "bury" })] }),
            }),
          }),
        },
      });
      const buriedPlayers = wrapper.findAllComponents<typeof GameOverHistoryRecordDecisionBuriedPlayer>(".game-over-history-record-buried-player");

      expect(buriedPlayers).toHaveLength(0);
    });
  });

  describe("Overflow Tag", () => {
    it("should display the overflow tag when rendered.", () => {
      const overflowTag = wrapper.findComponent<typeof OverflowTag>("#buried-players-overflow-tag");

      expect(overflowTag.props("entitiesCount")).toBe(4);
      expect(overflowTag.props("maximumEntitiesDisplayed")).toBe(3);
    });
  });
});