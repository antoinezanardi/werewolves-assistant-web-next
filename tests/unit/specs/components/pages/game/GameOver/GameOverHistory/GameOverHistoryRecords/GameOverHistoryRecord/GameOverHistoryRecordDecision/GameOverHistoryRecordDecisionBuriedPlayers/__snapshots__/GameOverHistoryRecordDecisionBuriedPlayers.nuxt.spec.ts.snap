// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Over History Record Decision Buried Players Component > should match snapshot when rendered. 1`] = `
"<div id="game-over-history-record-decision-buried-players" class="flex gap-4 items-center justify-center">
  <game-over-history-record-decision-buried-player-stub buriedplayer="[object Object]" gamehistoryrecord="[object Object]" class="game-over-history-record-buried-player w-full"></game-over-history-record-decision-buried-player-stub>
  <game-over-history-record-decision-buried-player-stub buriedplayer="[object Object]" gamehistoryrecord="[object Object]" class="game-over-history-record-buried-player w-full"></game-over-history-record-decision-buried-player-stub>
  <game-over-history-record-decision-buried-player-stub buriedplayer="[object Object]" gamehistoryrecord="[object Object]" class="game-over-history-record-buried-player w-full"></game-over-history-record-decision-buried-player-stub>
  <overflow-tag-stub entitiescount="4" maximumentitiesdisplayed="3" id="buried-players-overflow-tag"></overflow-tag-stub>
</div>"
`;
